src/gooso.ts

```diff
@@ -0,0 +1,64 @@
+import {
+  $effect,
+  $familiar,
+  $item,
+  $items,
+  $location,
+  $monster,
+  $skill,
+  getBanishedMonsters,
+  have,
+  getModifier
+} from "libram";
+import { haveEffect, retrieveItem, use } from "kolmafia";
+import { ChronerQuest, ChronerStrategy, ChronerTask } from "./engine";
+import Macro from "./macro";
+import { chooseQuestOutfit, ifHave } from "./outfit";
+
+const location = $location`Moonshiners' Woods`;
+const banishers = $items`ice house, human musk`;
+
+export const gooso: ChronerQuest = {
+  name: "Gooso",
+  location,
+  tasks: [
+    ...banishers.map(
+      (i): ChronerTask => {
+        return {
+          name: `Banisher ${i}`,
+          completed: () => have(i) || getBanishedMonsters().has(i),
+          do: () => retrieveItem(i),
+          sobriety: "sober",
+        };
+      }
+    ),
+    {
+      name: "Blue Taffy",
+      completed: () => haveEffect($effect`Blue Swayed`) < 50,
+      do: () => {
+        retrieveItem($item`pulled blue taffy`);
+        use($item`pulled blue taffy`);
+      },
+      sobriety: "either",
+    },
+    {
+      name: "Gooso",
+      completed: () => false,
+      do: location,
+      combat: new ChronerStrategy(
+        Macro.if_($monster`Moonshriner`, Macro.item($item`ice house`))
+          .if_($monster`unstill`, Macro.item($item`human musk`))
+          .trySkill($skill`Emit Matter Duplicating Drones`)
```

Comment:
I would add an if_ to this to avoid casting it against sausage goblins and romance wanderers